src/WCSimConstructCylinder.cc

```diff
@@ -932,6 +932,8 @@ G4LogicalVolume* WCSimDetectorConstruction::ConstructCaps(G4int zflip)
   G4double borderAnnulusZ[3] = {(-barrelCellHeight/2.-(WCIDRadius-innerAnnulusRadius))*zflip, 
 								-barrelCellHeight/2.*zflip,
 								barrelCellHeight/2.*zflip};
+  // to check edge case
+  if (WCBarrelPMTOffset < (WCIDRadius-innerAnnulusRadius)) borderAnnulusZ[0] = (-barrelCellHeight/2.-WCBarrelPMTOffset)*zflip;
```

Comment:
This is to make sure the [barrel ring](https://github.com/WCTE/WCSim/blob/288204e52e9654aad377887418797c2b50b8cc19/src/WCSimConstructCylinder.cc#L939) does not overlap with the [cap](https://github.com/WCTE/WCSim/blob/288204e52e9654aad377887418797c2b50b8cc19/src/WCSimConstructCylinder.cc#L1209)